src/Popper.js

```diff
@@ -154,22 +169,49 @@ export class Popper extends Component<PopperProps, PopperState> {
     }
   }
 
+  onPopperUpdate() {
+    this.state.subscribers.forEach(scheduleUpdate => scheduleUpdate());
+  }
+
+  subscribe = (scheduleUpdate: () => void) =>
+    this.setState(({ subscribers }) => ({
+      subscribers: [...subscribers, scheduleUpdate],
+    }));
+
+  unsubscribe = (scheduleUpdate: () => void) =>
+    this.setState(({ subscribers }) => ({
+      subscribers: subscribers.filter(
+        subscriber => subscriber !== scheduleUpdate
+      ),
+    }));
+
   componentWillUnmount() {
     if (this.state.popperInstance) {
       this.state.popperInstance.destroy();
     }
+    this.props.unsubscribe && this.props.unsubscribe(this.scheduleUpdate);
+  }
+
+  componentDidMount() {
+    this.props.subscribe && this.props.subscribe(this.scheduleUpdate);
   }
 
   render() {
-    return unwrapArray(this.props.children)({
-      ref: this.setPopperNode,
-      style: this.getPopperStyle(),
-      placement: this.getPopperPlacement(),
-      arrowProps: {
-        ref: this.setArrowNode,
-        style: this.getArrowStyle(),
-      },
-    });
+    return (
+      <PopperContext.Provider
+        value={{ subscribe: this.subscribe, unsubscribe: this.unsubscribe }}
```

Comment:
This will change context every render :)

Reply:
Right! Please feel free to fix it, it should be enough to add a class property to the component with the two methods as properties: